wgpu-types/src/lib.rs

```diff
@@ -1116,33 +1116,30 @@ pub struct PassChannel<V> {
     pub read_only: bool,
 }
 
-/// Describes a color attachment to a [`RenderPass`].
+/// Describes a color attachment to a render pass.
 #[repr(C)]
 #[derive(Clone, Debug)]
 #[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
-pub struct RenderPassColorAttachmentDescriptorBase<T> {
-    /// Texture attachment to render to. Must contain [`TextureUsage::OUTPUT_ATTACHMENT`].
+pub struct RenderPassColorAttachmentDescriptor<T, P> {
+    /// The view to use as an attachment.
     pub attachment: T,
-    /// MSAA resolve target. Must contain [`TextureUsage::OUTPUT_ATTACHMENT`]. Must be `None` if
-    /// attachment has 1 sample (does not have MSAA). This is not mandatory for rendering with multisampling,
-    /// you can choose to resolve later or manually.
+    /// The view that will receive the resolved output if multisampling is used.
     pub resolve_target: Option<T>,
-    /// Color channel.
-    pub channel: PassChannel<Color>,
+    /// What operations will be performed on this color attachment.
+    pub channel: P,
 }
 
-/// Describes a depth/stencil attachment to a [`RenderPass`].
+/// Describes a depth/stencil attachment to a render pass.
 #[repr(C)]
 #[derive(Clone, Debug)]
 #[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
-pub struct RenderPassDepthStencilAttachmentDescriptorBase<T> {
-    /// Texture attachment to render to. Must contain [`TextureUsage::OUTPUT_ATTACHMENT`] and be a valid
-    /// texture type for a depth/stencil attachment.
+pub struct RenderPassDepthStencilAttachmentDescriptor<T, P1, P2> {
+    /// The view to use as an attachment.
     pub attachment: T,
-    /// Depth channel.
-    pub depth: PassChannel<f32>,
-    /// Stencil channel.
-    pub stencil: PassChannel<u32>,
+    /// What operations will be performed on the depth part of the attachment.
+    pub depth: P1,
+    /// What operations will be performed on the stencil part of the attachment.
+    pub stencil: P2,
```

Comment:
The reason for using so many generic parameters is that `wgpu-core` has the `PassChannel` type, while `wgpu-rs` uses the `Operations` type. These types have a similar meaning but have different fields and structure.

Reply:
Is there a particular reason they can't be unified? If it's possible we probably should unify them.

Reply:
It is possible, but I don't know which is better: use the `Operations` structure in both places
```rust
pub struct Operations<V> {
    /// How data should be read through this attachment.
    pub load: LoadOp<V>,
    /// Whether data will be written to through this attachment.
    pub store: bool,
}
```
or the `PassChannel` one
```rust
pub struct PassChannel<V> {
    /// Operation to perform to the output attachment at the start of a renderpass. This must be clear if it
    /// is the first renderpass rendering to a swap chain image.
    pub load_op: LoadOp,
    /// Operation to perform to the output attachment at the end of a renderpass.
    pub store_op: StoreOp,
    /// If load_op is [`LoadOp::Clear`], the attachement will be cleared to this color.
    pub clear_value: V,
    /// If true, the relevant channel is not changed by a renderpass, and the corresponding attachment
    /// can be used inside the pass by other read-only usages.
    pub read_only: bool,
}
```

Reply:
Let's just not unify them. It's good to stick to `PassChannel` here in wgpu-core, because it's `repr(C)` and we can use it seamlessly by C clients, such as wgpu-native and Gecko.
However, `Operations` works really nice for `wgpu-rs`, so it should stay this way.